pkg/printers/internalversion/printers.go

```diff
@@ -1155,8 +1165,24 @@ func printJob(obj *batch.Job, options printers.GenerateOptions) ([]metav1.TableR
 	default:
 		jobDuration = duration.HumanDuration(obj.Status.CompletionTime.Sub(obj.Status.StartTime.Time))
 	}
+	var status string
+	if obj.ObjectMeta.DeletionTimestamp != nil {
+		status = "Terminating"
+	} else if hasJobCondition(obj.Status.Conditions, batch.JobComplete) {
+		status = "Complete"
+	} else if hasJobCondition(obj.Status.Conditions, batch.JobFailed) {
+		status = "Failed"
+	} else if hasJobCondition(obj.Status.Conditions, batch.JobSuspended) {
+		status = "Suspended"
+	} else if hasJobCondition(obj.Status.Conditions, batch.JobFailureTarget) {
+		status = "FailureTarget"
+	} else if obj.Status.StartTime == nil {
```

Comment:
Not really sure if we should have this one. Its probably fine but it is such a short lived status so not sure of the value.

Starting could be in a long state if the job is suspended and if the pod is waiting to be scheduled.

Reply:
Yeah, I agree, let's drop starting for now. IMO it might be confusing, because a natural interpretation from the user's point-of-view is that starting is when there are still pending pods.

Reply:
Should I consolidate this with "Failed", or just ignore this status?

Reply:
Ignore, this should consolidate it with Running, (startTime=nil) is normal at the beginning of each job run

Reply:
Done. I consolidated it with Running. I was confused with the previous comment (FailureTarget), so I asked to consolidate it with Failed. But it's all good now.